plex/player.py

```diff
@@ -174,10 +176,48 @@ def _startPlayback(self):
             )
             self._duration = toMilliseconds(self.getTotalTime())
 
+            # register settings
+            settings = self._mediaProvider.prepareSettings()
+            if not settings:
+                log('failed to load settings for {} ({}) playing from {}' \
+                    .format(self._item.title, self._file, mediaProvider2str(self._mediaProvider)), xbmc.LOGWARNING)
+                self._reset()
+                return
+
+            # load external subtitles
+            if settings.getBool(SETTINGS_PROVIDER_PLAYBACK_ENABLE_EXTERNAL_SUBTITLES):
+                self._addExternalSubtitles(plexServer.PlexServer())
+
         else:
             self._reset()
 
 
+    def _addExternalSubtitles(self, plexServer):
+        '''Add external subtitles to the player'''
+        if not self._item:
+            return
+
+        external_subtitles = self._item.subtitleStreams()
```

Comment:
Do we know if `subtitleStreams()` only returns external subtitles or all subtitles? If it's the latter can we distinguish between the two types? Otherwise we wend up with all "internal" subtitles appearing twice.

Reply:
Nice catch, I'll add a movie with internal subtitles to see

Reply:
You're right! The addon was trying to add internal subtitles too. For internal subtitles, `key` is `None` so I filter them out.

Reply:
Ah that makes sense because they can't be streamed separately.